BlazarService/src/main/java/com/hubspot/blazar/util/RepositoryBuildLauncher.java

```diff
@@ -90,6 +103,15 @@ private CommitInfo commitInfo(GitInfo gitInfo, Optional<Commit> previousCommit)
       throw new NonRetryableBuildException("Couldn't find repository " + gitInfo.getFullRepositoryName(), e);
     }
 
+    Optional<Commit> previousCommit = getCommitFromRepositoryBuild(previousBuild);
+
+    // Inter project builds re-build the project using the commit of the last build if available
+    // otherwise we fall back to using the newest commit available
+    if (currentBuild.getBuildTrigger().getType() == INTER_PROJECT && previousCommit.isPresent()) {
+      return gitHubHelper.commitInfoFor(repository, previousCommit.get(), previousCommit);
```

Comment:
why don't you just do:
`return new CommitInfo(lastCommitInPreviousBuild, lastCommitInPreviousBuild, ImmutableList.of(), false);`